src/wasm/wasm-stack.cpp

```diff
@@ -2089,6 +2089,24 @@ void BinaryInstWriter::visitRefTest(RefTest* curr) {
 }
 
 void BinaryInstWriter::visitRefCast(RefCast* curr) {
+  // We allow ref.cast of string views, but V8 does not. Work around that by
+  // emitting a ref.as_non_null (or nothing).
+  auto type = curr->type;
+  if (type.isRef()) {
+    auto heapType = type.getHeapType();
+    if (heapType == HeapType::stringview_wtf8 ||
+        heapType == HeapType::stringview_wtf16 ||
+        heapType == HeapType::stringview_iter) {
+      // We cannot cast string views to/from anything, so the input must also
+      // be a view.
+      assert(curr->ref->type.getHeapType() == heapType);
```

Comment:
I don't think this will handle casts to bottom correctly. If a string view is cast to `(ref none)`, we should just emit `(unreachable)` because that cast cannot possibly succeed. If a string view is cast to `(ref null none)`, I haven't thought of anything better than emitting `(ref.is_null) (if (result nullref) (then (ref.null none)) (else (unreachable))`.

Reply:
This isn't comprehensive atm, I guess, yeah, but it should handle all the things the fuzzer and the optimizer emit. I'm not sure if it's worth handling more things.